docs: add `release` tag

NicolasT opened the pull request:
This commit adds a `release` tag to the Sphinx project. Currently, when
set, `todo` elements in the documentation will *not* be rendered. This
is, however, not the default.

A related change is made to pass the `O` environment variable from the
`tox` invocation all the way down to arguments to `sphinx-build`. As
such, one can now run

```
O="-t release" tox -e docs
```

to create 'release' builds of the documentation.

NicolasT committed:
docs: add `release` tag

This commit adds a `release` tag to the Sphinx project. Currently, when
set, `todo` elements in the documentation will *not* be rendered. This
is, however, not the default.

A related change is made to pass the `O` environment variable from the
`tox` invocation all the way down to arguments to `sphinx-build`. As
such, one can now run

```
O="-t release" tox -e docs
```

to create 'release' builds of the documentation.

bert-e commented:
# Hello nicolast,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/docs-add-release-tag` into
`development/2.4`
* [w/2.5/improvement/docs-add-release-tag](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/docs-add-release-tag) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

lucieleonard reviewed (commented):
I would add the new command in README.md

slaperche-scality commented:
Why not have a release config by default and add a `dev` flag to enable TODOs and stuff?

Less risky if we generate the doc in a hurry and forget the flag, at least it'll be "clean" by default.

NicolasT commented:
> Why not have a release config by default and add a `dev` flag to enable TODOs and stuff?
> 
> Less risky if we generate the doc in a hurry and forget the flag, at least it'll be "clean" by default.

I'd like all non-customer builds (i.e., ReadTheDocs, open-source builds,...) to include these TODOs since we welcome contributions to fix these.

slaperche-scality commented:
Ha, we have no control on how RTD build the doc (i.e. can't switch the flag for RTD)?

NicolasT commented:
> Ha, we have no control on how RTD build the doc (i.e. can't switch the flag for RTD)?

Maybe we can (didn't check), but even if we can, I'd like any default build to still include the tags.

However. This does raise the question about the docs we include in a build (HTML in the UI, or PDF)...

NicolasT committed:
buildchain: build docs in release mode

See: https://github.com/scality/metalk8s/pull/2226#issuecomment-583455887

NicolasT commented:
I added a patch which ensures the docs are built using the `release` tag while building the ISO. Manual testing seems to validate this is working.

NicolasT commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/3b1298497623fc0f523faa0b8c54eb95e0edd0af) did not succeed in branch improvement/docs-add-release-tag.




*The following options are set:* **approve**

NicolasT committed:
docs: pin `six` version and update `tox` version in Docker build

The `virtualenv` version pulled in by `tox` isn't happy with the
old(ish) version of `six` as pulled into the container, breaking all
builds.

Pinning `six` and bumping `tox` instead.

bert-e commented:
# History mismatch



Merge commit #33e87cc7b97d4e57f58e3316acab5b2be1a0909c on the integration branch
`w/2.5/improvement/docs-add-release-tag` is merging a branch which is neither the current
branch `improvement/docs-add-release-tag` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/docs-add-release-tag` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

NicolasT commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/docs-add-release-tag` into
`development/2.4`
* [w/2.5/improvement/docs-add-release-tag](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/docs-add-release-tag) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/49f6701266ae487ecb3bc3ff116e29716f558c20) did not succeed in branch improvement/docs-add-release-tag.




*The following options are set:* **approve**

gdemonet commented:
/bypass_build_status

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **bypass_build_status, approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye nicolast.
